File: search-service/src/main/java/com/soccer_stats/search_service/controller/PlayerController.java
package com.soccer_stats.search_service.controller;

import com.soccer_stats.search_service.document.Player;
import com.soccer_stats.search_service.service.PlayerService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/search/player")
public class PlayerController {

    private final PlayerService playerService;

    public PlayerController(PlayerService playerService) {
        this.playerService = playerService;
    }
    @GetMapping("/listAll/player")
    public Iterable<Player> getAllPlayers()
    {
        return playerService.getPlayers();
    }
}

File: search-service-kt/src/main/kotlin/com/soccer_stats/search_service_kt/controller/PlayerController.kt
package com.soccer_stats.search_service_kt.controller

import com.soccer_stats.search_service_kt.document.Player
import com.soccer_stats.search_service_kt.service.PlayerService
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/v1/player")
class PlayerController(
    private val playerService: PlayerService
){


    @GetMapping("/all")
    fun getAllPlayer(): Iterable<Player>
    {
        return playerService.getAllPlayers()
    }
    @DeleteMapping
    fun deletallPlayer()
    {
        playerService.deleteAllPlayers()
    }
}